leg:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50 

  # Position Controllers
  joint1_position_controller:
    type: effort_controllers/JointPositionController
    joint: base_to_link1
    pid: {p: 10.0, i: 0, d: 1}

  joint2_position_controller:
    type: effort_controllers/JointPositionController
    joint: link1_to_link2
    pid: {p: 10.0, i: 0, d: 1}


  joint3_position_controller:
    type: effort_controllers/JointPositionController
    joint: link2_to_link3
    pid: {p: 10.0, i: 0, d: 1}

  #Trajectory Controller:
  Trajectory_Controller:
    type: "effort_controllers/JointTrajectoryController"
    joints:
      - base_to_link1
      - link1_to_link2
      - link2_to_link3

    gains: # Required because we're controlling an effort interface
      base_to_link1: {p: 10.0, i: 0, d: 1}
      link1_to_link2: {p: 10.0, i: 0, d: 1}
      link2_to_link3: {p: 10.0, i: 0, d: 1}

    action_monitor_rate: 1000

  # Effort Controllers -> Force control ------------------------
  joint1_effort_controller:
    type: effort_controllers/JointEffortController
    joint: base_to_link1

  joint2_effort_controller:
    type: effort_controllers/JointEffortController
    joint: link1_to_link2

  joint3_effort_controller:
    type: effort_controllers/JointEffortController
    joint: link2_to_link3


